renderer/tr_light.cpp

```diff
@@ -1156,31 +1158,32 @@ void R_AddDrawSurf( const srfTriangles_t *tri, const viewEntity_t *space, const
 			drawSurf->sort += eDef->parms.sortOffset;
 	}
 
-	// bumping this offset each time causes surfaces with equal sort orders to still
-	// deterministically draw in the order they are added
-	tr.sortOffset += 0.000001f;
-
-	// if it doesn't fit, resize the list
-	Sys_EnterCriticalSection( CRITICAL_SECTION_TWO );
-	if ( tr.viewDef->numDrawSurfs == tr.viewDef->maxDrawSurfs ) {
-		drawSurf_t	**old = tr.viewDef->drawSurfs;
-		int			count;
+	if (!deferred) {
```

Comment:
Is there any place which uses non-deferred case?
If yes, than can't it be reworked to also use deferred code path?

Also, right now you have this code in two places, which is a bad idea.

Reply:
Yes, unfortunately `R_AddDrawSurf` is used in several places, and I'm not familiar with all of them. So I opted for the `deferred` parameter to not have to touch all of those places for now. But yeah, I should probably extract the list resize into its own function.

Reply:
Ok, after looking at it again, it turns out 3 of 4 uses of `R_AddDrawSurf` resulted from recursive calls from within `R_AddDrawSurf`, so those were a no-brainer to switch to deferred. The only remaining usage is within GUI code, of which parts are also a recursive calls, but parts are not. And since GUI itself needs to be protected if run in parallel, this is tricky to move to the deferred path. So I'm leaving that for now, but I'm now also fairly sure that the remaining use of the non-deferred path is not run in parallel, so no synchronization is needed.